peekingduck/declarative_loader.py

```diff
@@ -180,12 +178,69 @@ def _edit_config(
                         f"Config for node {node_name} does not have the key: {key}"
                     )
                 else:
+                    if key == "detect_ids" and node_name in (
+                        "model.yolo",
+                        "model.efficientdet",
+                    ):
+                        key, value = self._change_labels_to_ids(node_name, key, value)
                     dict_orig[key] = value
                     self.logger.info(
                         f"Config for node {node_name} is updated to: '{key}': {value}"
                     )
         return dict_orig
 
+    def _change_labels_to_ids(
+        self, node_name: str, key: str, value: list
+    ) -> Tuple[str, list]:
+        """Process model.yolo or model.efficientdet detect_ids and check for labels to
+        be converted to ids, e.g. person to 0, car to 2
+
+        Args:
+            node_name (str): to determine if node is yolo or efficientdet as both
+                             lists of ids are different.
+            key (str): can only be "detect_ids", else error.
+            value (list): list of labels/ids for detection.
+                          If ids, do nothing.
+                          If labels, to be converted to ids.
+
+        Returns:
+            Tuple[str, list]: "detect_ids", list of sorted ids for object detection
+        """
+        assert key == "detect_ids", f"Key Error: expect detect_ids but got {key}"
```

Comment:
Is this necessary considering it only gets called during `if key == "detect_ids"`?

Reply:
Removed `assert` line to reduce code. Wanted to "future-proof" `_change_labels_to_ids` but given the restricted scope in which this method is called, it's ok to not do extra checks (for now).